R/utils.R

```diff
@@ -1248,3 +1248,111 @@ setMethod("reduce", "data.frame",
               rownames(ans) <- NULL
               ans
           })
+
+##' Select feature variables to be retained.
+##'
+##' @title Select feature variables of interest
+##'
+##' @param object An \code{MSnSet}, \code{MSnExp} or \code{OnDiskMSnExp}.
+##' 
+##' @param graphics A \code{logical} (default is \code{TRUE})
+##'     indicating whether a shiny application should be used if
+##'     available. Otherwise, a text menu is used. Ignored if \code{k}
+##'     is not missing.
+##' 
+##' @param fcol A \code{numeric}, \code{logical} or \code{character} of
+##'     valid feature variables to be passed directly.
+##' 
+##' @return For \code{selectFeatureData}: updated object containing only
+##'     selected feature variables.
+##' 
+##' @author Laurent Gatto
+##' 
+##' @examples
+##' 
+##' library("pRolocdata")
+##' data(hyperLOPIT2015)
+##' ## 5 first feature variables
+##' x <- selectFeatureData(hyperLOPIT2015, fcol = 1:5)
+##' fvarLabels(x)
+##' \dontrun{
+##' ## select via GUI
+##' x <- selectFeatureData(hyperLOPIT2015)
+##' fvarLabels(x)
+##' }
+##'
+##' ## Subset the feature data of an OnDiskMSnExp object to the minimal
+##' ## required columns
+##' f <- system.file("microtofq/MM14.mzML", package = "msdata")
+##' od <- readMSData(f, mode = "onDisk")
+##'
+##' ## what columns do we have?
+##' fvarLabels(od)
+##'
+##' ## Reduce the feature data data.frame to the required columns only
+##' od <- selectFeatureData(od, fcol = requiredFvarLabels(class(od)))
+##' fvarLabels(od)
+selectFeatureData <- function(object,
+                              graphics = TRUE,
+                              fcol) {
+    if (missing(fcol)) {
+        if (graphics) {
+            if (!requireNamespace("shiny"))
+                warning("The shiny package is required to use the graphical interface.")
```

Comment:
IMHO this should be `if (graphics && !requireNamespace("shiny"))`. Currently a warning is thrown if `shiny` is missing. Nevertheless the `.selectShinyFeatureData` is executed and will fail.

@lgatto `MSnbase` really suggests `shiny` just for `fcol` selection? `MSnbase` has already a lot of dependencies. Starting a browser and build a webpage for selecting some columns of a `DataFrame` is a little bit weird. I would suggest to remove this `shiny` part.

Reply:
We could move `shiny` from `Suggests` to `Enhances` and use your suggested `requireNamespace`. Most users will have `shiny` anyway installed.

Reply:
In some applications (mainly with `MSnSet` for spatial proteomics), large feature tables are really annoying, and a shiny app to easily select them is handy (we even have such a tab in our `pRolocGUI` apps). 

@jotsetung - I don't think that `Enhances` is correct - it would imply that `MSnbase` enhances `shiny` (the meaning of `Enhances` has been confused and is confusing). `Suggests` is the right place.

Reply:
OK, I'll put it back to `Suggests`.